notebooks/instructlab-knowledge/utils/illuminator/analysis.py

```diff
@@ -40,37 +41,37 @@ def summarize_tables(doc) -> Tuple[int, List[int]]:
 
     return num_tables, pages
 
-
-def analyze_pdf_with_docling(file_path) -> Dict[str, Union[int, List[Any], set]]:
+def convert_pdf_with_docling(file_path: str) -> DoclingDocument:
```

Comment:
Might be nice to rename this to `convert_to_docling_document since it's not pdf specific anymore. It would also be nice for this function to have an argument on whether or not the markdown should be saved or not. I'm not sure in every case we want that.